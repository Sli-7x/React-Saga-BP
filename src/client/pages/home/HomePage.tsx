import * as React from 'react';
import { connect } from 'react-redux';
import { ErrorBoundary } from '../../components/errors/ErrorBoundary';
import { Filters } from '../../components/filters/Filters';
import { loadFilters } from '../../store/filters/actions';
import { errorToast } from '../../components/ui/toast/Toast.action';

interface IHomePageProps {
  filters: any;
  dispatch: any;
}

interface IHomePageState {
  fetchData: any;
}

export class HomePage extends React.Component<IHomePageProps, IHomePageState> {
  constructor(props: IHomePageProps) {
    super(props);
    this.getList = this.getList.bind(this);
  }

  componentDidMount() {
    this.props.dispatch(loadFilters({ id: '5555' }));
  }

  getList() {
    this.props.dispatch(errorToast('ERRORRRORROROROR'));
    this.props.dispatch(loadFilters({ id: '5555' }));
  }

  render() {
    const { filters } = this.props;

    return (
      <div>
        <h1>Home page</h1>
        <button onClick={this.getList}>Get</button>
        <ErrorBoundary>{filters.loading ? <h3>Loading...</h3> : <Filters list={filters.list} />}</ErrorBoundary>
      </div>
    );
  }
}

export const mapStateToProps = (state: any) => ({
  filters: state.filters,
  toast: state.toast,
});

export default connect(mapStateToProps)(HomePage);
