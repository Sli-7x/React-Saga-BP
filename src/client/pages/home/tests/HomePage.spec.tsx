import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { HomePage, mapStateToProps } from '../HomePage';

const dispatch = jest.fn();

describe('HomePage.tsx', () => {
  it('should render correctly', () => {
    const filledTree = renderer.create(<HomePage filters={{}} dispatch={dispatch} />).toJSON();
    expect(filledTree).toMatchSnapshot();

    const filledTree2 = renderer.create(<HomePage filters={{ loading: true }} dispatch={dispatch} />).toJSON();
    expect(filledTree2).toMatchSnapshot();
  });

  it('should cover mapStateToProps', () => {
    expect(mapStateToProps({ filters: [] })).toEqual({ filters: [] });
  });
});
