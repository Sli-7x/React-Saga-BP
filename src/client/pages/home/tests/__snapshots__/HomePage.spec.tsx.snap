// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage.tsx should render correctly 1`] = `
<div>
  <h1>
    Home page
  </h1>
  <button
    onClick={[Function]}
  >
    Get
  </button>
  <div>
    <h2>
      Something went wrong.
    </h2>
    <details
      style={
        Object {
          "whiteSpace": "pre-wrap",
        }
      }
    >
      TypeError: Cannot read property 'map' of undefined
      <br />
      
    in Filters (created by HomePage)
    in ErrorBoundary (created by HomePage)
    in div (created by HomePage)
    in HomePage
    </details>
  </div>
</div>
`;

exports[`HomePage.tsx should render correctly 2`] = `
<div>
  <h1>
    Home page
  </h1>
  <button
    onClick={[Function]}
  >
    Get
  </button>
  <h3>
    Loading...
  </h3>
</div>
`;
